helios-services/src/main/java/com/spotify/helios/master/resources/HostsResource.java

```diff
@@ -122,11 +123,14 @@ public HostsResource(final MasterModel model) {
           })
           .collect(Collectors.toList());
 
+      final Function<String, Map<String, String>> getLabels =
+          host -> {
+            final HostStatus hostStatus = model.getHostStatus(host);
+            return hostStatus == null ? emptyMap() : hostStatus.getLabels();
```

Comment:
would it be easier if `model.getHostStatus(host)` returned an `Optional<String>` instead? seems like every place that calls that method has to perform the null check, so might be nice to address that with the type system itself